### ISSUE TITLE
Not able to spawn a consumer using tokio core handle

### ISSUE THREAD

**Issue by surajpayu** [fede1024/rust-rdkafka#83]
I already have core instance available. I want to spawn a consumer using handle of core

```rust
  let processed_stream = consumer.start();
     let h = processed_stream.filter_map(|result| {  // Filter out errors
            match result {
                Ok(msg) => Some(msg),
                Err(kafka_error) => {
                    println!("Error while receiving from Kafka: {:?}", kafka_error);
                    None
                }
            }
        }).for_each(|msg| {     // Process each message
            println!("Enqueuing message for computation");
            let owned_message = msg.detach();
            let t = match owned_message.payload_view::<str>() {
                Some(Ok(payload)) => String::from(payload),
                Some(Err(_)) => "Error processing message payload".to_owned(),
                None => "No payload".to_owned(),
            };
            let k = match owned_message.key_view::<str>() {
                Some(Ok(key)) => String::from(key),
                Some(Err(_)) => "Error processing message key".to_owned(),
                None => "No key".to_owned(),
            };
            // Create the inner pipeline, that represents the processing of a single event.
            // Spawns the inner pipeline in the same event pool.
            //let d = http_call(3, 4, owned_message);
            //println!("Response is {}", d);
            // http_call(owned_message);
            // load_merchant_script(String::from(input_topic), t);
             let tx = HASHMAP.write().unwrap().remove(&k[..]).unwrap();
             println!("Consumer cp4");
             tx.send(t).unwrap();
            // println!("Message is {}",t);
            Ok(())
        });
    println!("Starting event loop");
    // Runs the event pool until the consumer terminates.
    //handle.spawn(processed_stream);
    handle.spawn(h);
```

I am getting below error
```
   Compiling payu_dsl_core v0.1.0 (file:///home/suraj.prakash/projects/payu_dsl_core)
warning: unnecessary parentheses around `if` condition
    --> main.rs:1045:19
     |
1045 |                 if(f > max){max = f;st.clear();st.push_str(&st2i[..]);}
     |                   ^^^^^^^^^
     |
     = note: #[warn(unused_parens)] on by default

error[E0597]: `consumer` does not live long enough
   --> main.rs:943:28
    |
943 |     let processed_stream = consumer.start();
    |                            ^^^^^^^^ does not live long enough
...
982 | }
    | - borrowed value only lives until here
```

**Comment by fede1024** [comment]
Currently a stream generated by the consumer holds a reference to the consumer itself, so the consumer should live as long as the stream does.

**Comment by surajpayu** [comment]
can we please remove this dependency so that i can use my core in more optimum manner? I want to go live with your library in production but because of this, i am blocked.

**Comment by fede1024** [comment]
Could you try actually with:
```
let processed_stream = consumer.start()
       .filter_map( ... ) ... ;

core.run(processed_stream).unwrap();
```

**Comment by surajpayu** [comment]
No, actually i am creating a hyper http server and i wanted to use its handle for running the kafka consumer stream.  Would it be possible for you to fix this? Why do we need the stream to hold a reference to consumer. I think we should de-link them. Such cross dependency restricts its usage.

**Comment by surajpayu** [comment]
Please update

**Comment by fede1024** [comment]
When the consumer falls out of scope the internal thread is terminated. Unfortunately I'm very busy at the moment and I can't say when I'll be able to follow up on this. The best way to go is probably to have the `MessageStream` own `StreamConsumer` directly instead of having a reference to it. PR are welcome.

**Comment by surajpayu** [comment]
I can wait for some time, could you change it in next one or two months. I will hit production in two months. Please let me know.

**Comment by fede1024** [comment]
I had some time to look into this and unfortunately it doesn't seem to be possible to easily share the `MessageStream` across threads, dues to the lifetime of the consumer. See my [comment](https://github.com/fede1024/rust-rdkafka/issues/85#issuecomment-398280136) on issue #85.

**Comment by benesch** [comment]
Closing due to age, as the entire async/await ecosystem has been completely rewritten since this issue was filed. I *think* what you want is now possible, but it's hard to say for sure given the amount of churn in tokio/futures. Please feel free to file a new issue if you are unable to achieve what you want with the latest master (to be released as 0.23).
